Common/Data/Custom/Benzinga/BenzingaNewsFactory.cs

```diff
@@ -0,0 +1,226 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using QuantConnect.Data.Auxiliary;
+using QuantConnect.Interfaces;
+using QuantConnect.Logging;
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Text.RegularExpressions;
+using static QuantConnect.StringExtensions;
+
+namespace QuantConnect.Data.Custom.Benzinga
+{
+    /// <summary>
+    /// Benzinga news data deserialization utility methods - https://docs.benzinga.io/benzinga/newsfeed-v2.html
+    /// </summary>
+    public static class BenzingaNewsFactory
+    {
+        /// <summary>
+        /// Sometimes "Berkshire Hathaway" is mentioned as "BRK" in the raw data, although it is
+        /// separated into class A and B shares and should appear as BRK.A and BRK.B. Because our
+        /// map file system does not perform the conversion from BRK -> { BRK.A, BRK.B }, we must
+        /// provide them manually. Note that we don't dynamically try to locate class A and B shares
+        /// because there can exist companies with the same base ticker that class A and B shares have.
+        /// For example, CBS trades under "CBS" and "CBS.A", which means that if "CBS" appears, it will
+        /// be automatically mapped to CBS. However, if we dynamically selected "CBS.A" - we might select
+        /// a different company not associated with the ticker being referenced.
+        /// </summary>
+        public static readonly Dictionary<string, HashSet<string>> ShareClassMappedTickers = new Dictionary<string, HashSet<string>>
+        {
+            {"BRK", new HashSet<string>
+                {
+                    "BRK.A",
+                    "BRK.B"
+                }
+            }
+        };
+
+        /// <summary>
+        /// Creates an instance of <see cref="BenzingaNews"/> from RSS data (historical)
+        /// </summary>
+        /// <param name="contents">Raw contents of the XML file</param>
+        /// <param name="mapFileProvider">Map file provider</param>
+        /// <param name="mapFileResolver">Map file resolver</param>
+        /// <returns>BenzingaNews instance</returns>
+        public static BenzingaNews CreateBenzingaNewsFromRSS(string contents, IMapFileProvider mapFileProvider, MapFileResolver mapFileResolver)
```

Comment:
I was faced with the option of either creating an XML schema file and parsing it using that (with xsd.exe which outputs a VERY ugly class and does not follow project style guidelines), or dealing with it the way I have it, which is to take the XML, and convert it to JSON before parsing it. Since I opted out from creating a data type specifically for the RSS feed (to keep project bloat to a minimum), I dynamically parsed it from a JObject.

Unfortunately, I understand that it gets super complex, especially given that no schema for the data was provided. My new commit will contain the schema of the data we're parsing for maintainability purposes and get rid of the RSS parsing stuff 👍